api/views.py

```diff
@@ -247,6 +248,55 @@ def get_queryset(self):
             guest__status='Accepted')
         return open_games_as_guest.order_by("date","time")
 
+class MyGamesList(ListAPIView):
+    serializer_class = GameSessionSerializer
+    permission_classes = [permissions.IsAuthenticated,]
+
+    def get_queryset(self):
+        my_games = GameSession.objects.filter(
+            date__gte=datetime.now(pytz.timezone('America/New_York')))
+    
+        my_games_search = self.request.query_params.get("my-games")
+        if my_games_search is not None:
+            #User's Confirmed Games as Guest and Host
+            if my_games_search == "AllConfirmed":
+                my_games = my_games.filter(confirmed=True)
+                my_host_confirmed_games =  my_games.filter(host=self.request.user)
+                my_guest_confirmed_games = my_games.filter(guest__user=self.request.user, guest__status='Accepted')
```

Comment:
views.py 265, the guest__status='Accepted' that is user as guest instance or guest, right? I don't know whether you tested for this, bc it would be a bit nasty of a bug if not.